src/conv-core/conv-rdmadevice.h

```diff
@@ -0,0 +1,75 @@
+#ifndef _CONV_RDMADEVICE_H_
+#define _CONV_RDMADEVICE_H_
+
+#if CMK_CUDA
+
+#include "cmirdmautils.h"
+#include "pup.h"
+#include <cuda_runtime.h>
+
+// Represents the mode of device-side zerocopy transfer
+// MEMCPY indicates that the PEs are on the same logical node and cudaMemcpyDeviceToDevice can be used
+// IPC indicates that the PEs are on different logical nodes within the same physical node and CUDA IPC can be used
+// RDMA indicates that the PEs are on different physical nodes and requires GPUDirect RDMA
+enum class CmiNcpyModeDevice : char { MEMCPY, IPC, RDMA };
+
+class CmiDeviceBuffer {
+public:
+  // Pointer to and size of the buffer
+  const void* ptr;
+  size_t cnt;
+
+  // Home PE
+  int pe;
+
+  // Used for CUDA IPC
+  int device_idx;
+  size_t comm_offset;
+  int event_idx;
+  cudaStream_t cuda_stream;
```

Comment:
Just confirming that it is not required to pup cuda_stream?

Reply:
Correct, `cuda_stream` in `CmiDeviceBuffer` is only used on the sender side in `CkRdmaDeviceOnSender` in `ckrdmadevice.C`.
On the receiver side the `cuda_stream` from the user-specified `CkDeviceBufferPost` is used, in `CkRdmaIssueRgets` which is also in `ckrdmadevice.C`.
Thanks for checking on this though.